lib/hashtable.rb

```diff
@@ -0,0 +1,35 @@
+require_relative './hash_linked_list'
+require 'digest'
+
+class Hashtable
+  attr_reader :size
+
+  def initialize(tableSize)
+    @table = Array.new(tableSize).map!{ |bucket| LinkedList.new }
+    @size = 0
+    @TABLESIZE = tableSize
+  end
+
+  def set(key, value)
+    raise "Need a string" unless key.is_a?(String)
```

Comment:
Since you aren't planning on rescuing this, you can listen to rubocop, and make it a `fail`